crates/core/src/manifest.rs

```diff
@@ -130,32 +135,40 @@ impl Manifest {
     #[tracing::instrument(skip_all)]
     pub fn save(&self) -> Result<(), ProtoError> {
         use fs4::FileExt;
-        use std::io::prelude::*;
+        use std::io::{prelude::*, SeekFrom};
 
         debug!(file = %self.path.display(), "Saving manifest");
 
         if let Some(parent) = self.path.parent() {
             fs::create_dir_all(parent)?;
         }
 
-        let mut file = fs::create_file(&self.path)?;
```

Comment:
This would truncate the file, and then other processes would read an empty file causing failures.